cmd/pj-rehearse/main.go

```diff
@@ -226,4 +243,16 @@ func main() {
 		gracefulExit(o.noFail, jobsFailureOutput)
 	}
 	logger.Info("All jobs were rehearsed successfully")
+
+	logger.Info("Cleanup temporary configMaps")
+	if err := cmCleanup(cmController); err != nil {
```

Comment:
Placing the cleanup here will still only clean up the CMs when no problems occur. All the places between CM creation and here that call `gracefulExit` for different reasons will cause `pj-rehearse` to leak CMs on the cluster, which I believe we should not do, it's a resource leak.